code/game/objects/items/holotool.dm

```diff
@@ -0,0 +1,114 @@
+/obj/item/holotool
+	name = "experimental holotool"
+	desc = "A highly experimental holographic tool projector."
+	icon = 'icons/obj/objects.dmi'
+	inhand_icon_state = "holotool"
+	icon_state = "holotool"
+	damtype = BURN
+	force = 0
+	slot_flags = ITEM_SLOT_BELT
+	hitsound = 'sound/weapons/blade1.ogg'
+	usesound = 'sound/items/pshoom.ogg'
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	actions_types = list(/datum/action/item_action/change_tool, /datum/action/item_action/change_ht_color)
+	w_class = WEIGHT_CLASS_TINY
+
+	var/obj/item/current_tool = 0
+	var/current_color = "#48D1CC" //mediumturquoise
+
+/obj/item/holotool/proc/AddTool(typee, namee)
+	var/obj/item/WR = new typee(src)
+	WR.forceMove(src)
+	WR.name = namee
+	WR.usesound = usesound //use the same sound as we do
+	WR.toolspeed = 0.55
```

Comment:
setting toolspeed like this doesn't work, you need to put it in the variables of the holotool.